modules/openapi-generator/src/main/java/org/openapitools/codegen/utils/ModelUtils.java

```diff
@@ -1379,6 +1379,14 @@ public static Schema unaliasSchema(OpenAPI openAPI,
                 return schema;
             }
             Schema ref = allSchemas.get(simpleRef);
+
+            // override the description of the referenced component
+            if(ref != null) {
+                if(schema.getDescription() != null) {
+                    ref.setDescription(schema.getDescription());
+                }
+            }
+
```

Comment:
thanks for the fix

but this change will permanently update the description of the reference schema. 

if the reference schema is used somewhere else, the description will be the overridden one instead of the original one.

i think a better way to fix it is in the `fromProperty` in default codegen. or maybe using normalizer to fix it.

I'll try to find time to come up with a fix later this week

Reply:
`fromProperty` what do you mean? Please clarify, maybe I can update the approach